Release - Jobs DTO (#622)

HayenNico opened the pull request:
## Description

Update for the DTO instances for the new jobs coming with release 4.4.

## Motivation

Jobs can be very freely defined using the job configurations, so the DTOs only really need two parameters: the job type (to grab the proper definition from config) and the job parameters (filled in instance of the  "template" field in the job config).
Some additional optional parameters are suggested that may apply in common use cases.

## Fixes:

src/jobs/dto/create-job.dto.ts
* fix: jobParams parameter no longer optional

## Changes:

src/jobs/dto/create-job.dto.ts
* new: optional parameters id, dependsOn, contactEmail
* added swagger decorators

src/jobs/dto/update-job.dto.ts
* new: parameter id is required in this case
* added swagger decorators

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

Junjiequan committed:
fix: updated mathJS package version && improved number check

Junjiequan committed:
fix: updated mathJS package version && improved number check

HayenNico committed:
Initial suggestions #622

CreateJobDTO and UpdateJobDTO with API decorators etc.. Besides basic functionality already required in the controller some additional suggestions for properties.

HayenNico committed:
fix: removed unnecessary imports

HayenNico committed:
Merge branch 'master' into Rel-4-4-Issue-622

HayenNico commented:
Most tests fail since jobs.schema not yet implemented - will need resolution of #621 first, possibly others

bpedersen2 commented:
Needs rebase due to #854 


There some eslint complaints that should get  fixed as well.

bpedersen2 commented on src/jobs/dto/create-job.dto.ts:
```diff
@@ -29,4 +24,36 @@ export class CreateJobDto {
+  @ApiProperty({
+    type: String,
+    required: false,
+    description: "Id for the job to be crated.",
```
s/crated/created/

HayenNico committed:
fix eslint complaints

HayenNico committed:
fix eslint complaints

HayenNico committed:
fix typo in job id decription

HayenNico committed:
fix typo in job id decription

bpedersen2 commented:
Note that we don't run any checks against non-master branches and this would only go  to release-4.4 

I recommend to add new features always to  master and fork off release branches only for bug-fixing.

HayenNico committed:
Merge branch 'Rel-4-4-Issue-622' of https://github.com/SciCatProject/scicat-backend-next into Rel-4-4-Issue-622

nitrosx commented on src/jobs/dto/create-job.dto.ts:
```diff
@@ -1,32 +1,63 @@
 
+  /*
+  New stuff from here
+  */
```
Remove it

HayenNico replied:
Done. I just left that in for review because I wasn't sure we needed some of those additional properties (yet).

nitrosx commented on src/jobs/dto/update-job.dto.ts:
```diff
@@ -1,4 +1,14 @@
+  })
+  @IsString()
+  readonly id: string;
+}
```
I'm not sure if this is correct and needed, as the update on job is not implemented. 
We are going to have statusUpdate. 
Check issue #622 for more info

HayenNico replied:
Fixed. I've removed job-update.dto entirely, and added the status update instead.

HayenNico committed:
replace update-job.dto with update-jobstatus.dto

Removed update-job.dto as it will not be implemented.
Added job-statusupdate.dto.
